projects/storefrontlib/cms-components/product/carousel/product-carousel-item/product-carousel-item.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { Product } from '@spartacus/core';
+import {
+  ProductListItemContext,
+  ProductListItemContextSource,
+} from '../../product-list';
+
+@Component({
+  selector: 'cx-product-carousel-item',
+  templateUrl: './product-carousel-item.component.html',
+  providers: [
+    ProductListItemContextSource,
+    {
+      provide: ProductListItemContext,
+      useExisting: ProductListItemContextSource,
+    },
+  ],
+})
```

Comment:
As a convention we use `changeDetection: ChangeDetectionStrategy.OnPush,` on components.